tests/data_context/test_data_context.py

```diff
@@ -1607,6 +1611,62 @@ def test_get_checkpoint_raises_error_on_missing_batch_kwargs(empty_data_context)
         context.get_checkpoint("foo")
 
 
+# TODO: add more test cases
+def run_checkpoint_newstyle(titanic_pandas_multibatch_data_context_with_013_datasource):
+    context = titanic_pandas_multibatch_data_context_with_013_datasource
+    # add checkpoint config
+    checkpoint_config = CheckpointConfig(
+        config_version=1,
+        name="my_checkpoint",
+        run_name_template="%Y-%M-foo-bar-template",
+        expectation_suite_name="my_expectation_suite",
+        action_list=[
+            {
+                "name": "store_validation_result",
+                "action": {
+                    "class_name": "StoreValidationResultAction",
+                },
+            },
+            {
+                "name": "store_evaluation_params",
+                "action": {
+                    "class_name": "StoreEvaluationParametersAction",
+                },
+            },
+            {
+                "name": "update_data_docs",
+                "action": {
+                    "class_name": "UpdateDataDocsAction",
+                },
+            },
+        ],
+        validations=[
+            {
+                "batch_request": {
+                    "datasource_name": "titanic_multi_batch",
+                    "data_connector_name": "my_data_connector",
+                    "data_asset_name": "Titanic_1911",
+                }
+            }
+        ],
+    )
+    checkpoint_config_key = ConfigurationIdentifier(
+        configuration_key=checkpoint_config.name
+    )
+    context.checkpoint_store.set(key=checkpoint_config_key, value=checkpoint_config)
+
+    with pytest.raises(DataContextError, match=r"expectation_suite .* not found"):
```

Comment:
How is this test different than the one in test_checkpoint.py?

Reply:
This test uses `data_context.run_checkpoint`. The one in `test_checkpoint.py` calls `run` directly on an instantiated Checkpoint